controllers/machinehealthcheck_controller.go

```diff
@@ -258,17 +279,12 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 	logger.V(3).Info(
 		"Remediations are allowed",
 		"total target", totalTargets,
-		"max unhealthy", m.Spec.MaxUnhealthy,
+		unhealthyLimitKey, unhealthyLimitValue,
 		"unhealthy targets", len(unhealthy),
 	)
 
-	maxUnhealthy, err := getMaxUnhealthy(m)
-	if err != nil {
-		return ctrl.Result{}, errors.Wrapf(err, "Failed to get value for maxUnhealthy")
-	}
-
 	// Remediation is allowed so maxUnhealthy - unhealthyMachineCount >= 0
```

Comment:
Fixed it, although I believe there's too much coupling between mhc and short circuiting code. We probably need to abstract away the short circuiting code so that it's easier to add new conditions.